src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -231,6 +232,19 @@ else if (shouldStartLine && !isOnStartOfLine(details, targetSourceLine)) {
     }
 
     /**
+     * Checks that a right curly should be alone on line or alone with SEMI token is LAMBDA.
+     * @param bracePolicy option for placing the right curly brace
+     * @param details Details for validation
+     * @return true if a right curly should be alone on line or with semi if token is LAMBDA.
+     */
+    private static boolean shouldBeAlone(RightCurlyOption bracePolicy, Details details) {
```

Comment:
I agree. I unated their together in a separate method because in the method validate() already is immpossible add a new condition by reason of "cyclomatic complexity".